service/api_logic/login_logic.py

```diff
@@ -0,0 +1,111 @@
+from flask import current_app, url_for
+from flask_mail import Message
+from itsdangerous import URLSafeTimedSerializer
+from dto.api_output import OutPutUser
+from database.models import User
+from exept.handle_exeptions import handle_exceptions
+import bcrypt
+from logger.logger import Logger
+
+api_logic_logger = Logger("api_logic_logger", "api_logic_logger.log")
+
+def message_to_user_gmail(user: User, reset_url):
+    return  f"""
+    <html>
+        <body style="font-family: Arial, sans-serif; line-height: 1.6; color: #333; background-color: #f9f9f9; padding: 20px;">
+            <div style="max-width: 600px; margin: 0 auto; background: white; border: 1px solid #ddd; border-radius: 10px; padding: 20px;">
+                <h2 style="color: #4CAF50; text-align: center;">Password Reset Request</h2>
+                <p>Dear <strong>{user.username}</strong>,</p>
+                <p>You have requested to reset your password. To do so, click the button below:</p>
+                <div style="text-align: center; margin: 20px 0;">
+                    <a href="{reset_url}" style="display: inline-block; background-color: #4CAF50; color: white; text-decoration: none; padding: 10px 20px; border-radius: 5px; font-size: 16px;">Reset Your Password</a>
+                </div>
+                <p>If the button above doesn't work, copy and paste the following link into your browser:</p>
+                <p style="word-break: break-word;">
+                    <a href="{reset_url}" style="color: #4CAF50;">{reset_url}</a>
+                </p>
+                <p>If you did not request a password reset, please ignore this email or contact support.</p>
+                <hr style="border: none; border-top: 1px solid #ddd; margin: 20px 0;">
+                <p style="text-align: center; color: #777; font-size: 12px;">
+                    Best regards,<br>
+                    <strong>Your QSport Team</strong>
+                </p>
+            </div>
+        </body>
+    </html>
+    """
+
+
+@handle_exceptions
+@api_logic_logger.log_function_call()
+class UserService:
+    def __init__(self, user_dal):
+        self.user_dal = user_dal
+        self.s = URLSafeTimedSerializer(current_app.secret_key)
+
+
+    def create_user(self, email_front, username_front, password_front):
+        existing_user = self.user_dal.get_user_by_email_or_username(email_front, username_front)
+        if existing_user:
+            return OutPutUser().dump(existing_user)
+
+        salt = bcrypt.gensalt()
+        hashed_password = bcrypt.hashpw(password_front.encode('utf-8'), salt)
+
+        new_user = User(email=email_front, username=username_front, password_hash=hashed_password.decode('utf-8'))
+        self.user_dal.create_user(new_user)
+
+        return OutPutUser().dump(new_user)
+
+    def request_password_reset(self, email: str):
+        existing_user = self.user_dal.get_user_by_email(email)
+        if not existing_user:
+            return OutPutUser().dump(None)
```

Comment:
Rename to OutputUser (output is a single word, not a combination of two. A little misleading)